code/modules/projectiles/guns/ballistic/launchers.dm

```diff
@@ -57,55 +57,20 @@
 	casing_ejector = FALSE
 	weapon_weight = WEAPON_HEAVY
 	bolt_type = BOLT_TYPE_NO_BOLT
-	magazine_wording = "rocket"
+	internal_magazine = TRUE
+	cartridge_wording = "rocket"
 	empty_indicator = TRUE
 
-/obj/item/gun/ballistic/rocketlauncher/handle_atom_del(atom/A)
-	if(A == chambered)
-		chambered = null
-		if(!QDELETED(magazine))
-			QDEL_NULL(magazine)
-	if(A == magazine)
-		magazine = null
-		if(!QDELETED(chambered))
-			QDEL_NULL(chambered)
-	update_icon()
-	return ..()
-
-/obj/item/gun/ballistic/rocketlauncher/can_shoot()
-	return chambered?.BB
-
 /obj/item/gun/ballistic/rocketlauncher/unrestricted
 	pin = /obj/item/firing_pin
 
-/obj/item/gun/ballistic/rocketlauncher/process_chamber()
-	if(chambered)
-		chambered = null
-	if(magazine)
-		QDEL_NULL(magazine)
-	update_icon()
+/obj/item/gun/ballistic/rocketlauncher/afterattack()
+	. = ..()
+	magazine.get_round(FALSE) //Hack to clear the mag after it's fired
 
 /obj/item/gun/ballistic/rocketlauncher/attack_self_tk(mob/user)
 	return //too difficult to remove the rocket with TK
 
-/obj/item/gun/ballistic/rocketlauncher/attack_self(mob/living/user)
```

Comment:
I'm going to assume not since you removed this...

Reply:
Yes, it uses the base weapon handling stuff instead of it's own snowflake code now though.